File: Valid Parentheses/Valid Parentheses.java
public class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<Character>();
        char[] pars = s.toCharArray();
        for(char c : pars){
            switch(c){
                case '(':
                    stk.push(c);
                    break;
                case ')':
                    if(stk.empty() || stk.peek() != '('){
                        return false;
                    }
                    stk.pop();
                    break;
                case '[':
                    stk.push(c);
                    break;
                case ']':
                    if(stk.empty() || stk.peek() != '['){
                        return false;
                    }
                    stk.pop();
                    break;
                case '{':
                    stk.push(c);
                    break;
                case '}':
                    if(stk.empty() || stk.peek() != '{'){
                        return false;
                    }
                    stk.pop();
                    break;
                default:
                    return false;
            }
        }
        
        return stk.empty();
    }
}

File: Valid Parentheses/Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {

        if(s.empty())       return false;
        
        std::stack<char>    left;
        for(unsigned int i = 0; i < s.size(); ++i){
            switch(s[i]){
                case '(':
                    left.push(s[i]);
                    break;
                case ')':
                    if(left.empty() || left.top() != '(')
                        return false;
                    else
                        left.pop();
                    break;
                case '[':
                    left.push(s[i]);
                    break;
                case ']':
                    if(left.empty() || left.top() != '[')  
                        return false;
                    else
                        left.pop();
                    break;
                case '{':
                    left.push(s[i]);
                    break;
                case '}':
                    if(left.empty() || left.top() != '{')
                        return false;
                    else
                        left.pop();
                    break;
                default:
                    break;
            }
        }
        if(left.empty())                    return true;
        else                                return false;
    }
};
